rocev2/test/src/scala/rdma/RecvQTest.scala

```diff
@@ -0,0 +1,183 @@
+package rdma
+
+import org.scalatest.funsuite.AnyFunSuite
+import rdma.RdmaConstants.PSN_WIDTH
+import spinal.core._
+import spinal.core.sim._
+import spinal.lib._
+
+import scala.collection.mutable
+import scala.util.Random
+
+class SeqOutTest extends AnyFunSuite {
+
+  class RxDriver(
+      rx: Stream[Fragment[RdmaDataBus]],
+      clockDomain: ClockDomain,
+      opCode: OpCode.OpCode,
+      fragments: Int
+  ) {
+    val psnQueue = new mutable.Queue[Int]
+    rx.valid #= false
+    rx.last #= false
+    rx.bth.opcodeFull #= opCode.id
+
+    def push(psnList: List[Int]): Unit = {
+      psnQueue ++= psnList
+    }
+
+    def clear(): Unit = {
+      psnQueue.clear()
+    }
+
+    val driver = fork {
+      while (true) {
+        if (psnQueue.nonEmpty) {
+          rx.bth.psn #= psnQueue.dequeue()
+          rx.valid #= true
+          for (i <- 1 to fragments) {
+            if (i == fragments) {
+              rx.last #= true
+            }
+            clockDomain.waitSamplingWhere(rx.ready.toBoolean)
+            rx.last #= false
+          }
+        } else {
+          rx.valid #= false
+          clockDomain.waitRisingEdge()
+        }
+      }
+    }
+  }
+
+  class PsnSetter(
+      qpAttr: QpAttrData,
+      qpAttrUpdate: QpAttrUpdateNotifier,
+      clockDomain: ClockDomain
+  ) {
+    def set(psn: Int): Unit = {
+      qpAttr.epsn #= psn
+      qpAttrUpdate.pulseRqPsnReset #= true
+      clockDomain.waitRisingEdge()
+      qpAttrUpdate.pulseRqPsnReset #= false
+    }
+  }
+
+  class PsnGenerator {
+    var PSN: Int = -1
```

Comment:
`val psn`, use camel case style,
names with all capital letters represent constants,
do not use var.